### PR TITLE
[ONNX] Enable export of __xor_

### PR DESCRIPTION
**Pull request by neginraoof**
Enable export of ```__xor_```

### CODE REVIEW

**Review by hwangdeyu — APPROVED** [review]
LGTM

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -1183,6 +1183,11 @@ def __or_(g, input, other):
     return g.op("Or", input, other)
 
 
+@wrap_logical_op_with_cast_to_and_from("Bool")
+def __xor_(g, input, other):
+    return g.op("Xor", input, other)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:1188]: I think we don't export cases for inputs with "non-boolean values". For example:

```
>>> import torch
>>> a = torch.tensor([12])
>>> b = torch.tensor([10])
>>> a ^ b
tensor([6])
```

perhaps should put warning or error out in this case?

**Review by BowenBao — CHANGES_REQUESTED** [review]

File: torch/onnx/symbolic_opset9.py (hunk 1 above)
Comment (hwangdeyu) [reply torch/onnx/symbolic_opset9.py:1188]: @BowenBao  Any thing wrong with this example?

Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py:1188]: Yes, all inputs are casted to booleans: using @wrap_logical_op_with_cast_to_and_from("Bool")

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -1173,14 +1177,31 @@ def le(g, input, other):
     return gt_impl(g, input, other)
 
 
-@wrap_logical_op_with_cast_to_and_from("Bool")
 def __and_(g, input, other):
-    return g.op("And", input, other)
+    if input.type().scalarType() == "Bool" and \
+            other.type().scalarType() is not None and other.type().scalarType() == "Bool":
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:1182]: nit: why need both `other.type().scalarType() is not None` and `other.type().scalarType() == "Bool"`?

**Review by BowenBao — APPROVED** [review]
LGTM! minor comment

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 3]
```diff
@@ -4166,6 +4166,16 @@ def _test_compare_ops(self, model, num_inputs):
             self.run_test(model, x_float)
             self.run_test(model, x_int)
 
+    @skipIfUnsupportedMinOpsetVersion(9)
+    def test_and_or_xor(self):
+        class MyModel(torch.nn.Module):
+            def forward(self, x, y):
+                return x ^ y, x | y, x & y
```
Comment (hwangdeyu) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:4173]: nit: add the bitwise_not `~` test too.

**Pull request merged**